src/reducer/playlistReducer.js

```diff
@@ -0,0 +1,26 @@
+export const playlistReducer = (state, { type, payload }) => {
+  switch (type) {
+    case "SET_PLAYLISTS":
+      return {
+        ...state,
+        playlists: payload,
+      };
+    case "UPDATE_PLAYLIST":
+      return {
+        ...state,
+        playlists: state.playlists.map((pl) => {
+          if (pl._id === payload._id) {
+            return payload;
+          }
+          return pl;
+        }),
```

Comment:
can use ternary operator here

Reply:
missed a lot of things in hurry, thank for pointing out. will change